pandas/tests/extension/decimal/test_decimal.py

```diff
@@ -274,7 +274,18 @@ def test_dataframe_constructor_with_dtype():
     tm.assert_frame_equal(result, expected)
 
 
-@pytest.mark.parametrize("frame", [True, False])
+@pytest.mark.parametrize(
+    "frame",
+    [
+        pytest.param(
+            True,
+            marks=pytest.mark.xfail(
+                reason="pd.concat call inside NDFrame.astype reverts the dtype"
```

Comment:
@jorisvandenbossche any idea why this might happen?

Reply:
Debugged a bit, and turns out that it is the constructor in concat that looses it, and that's because you now copy EAs in the constructor, and so it's DecimalArray.copy() that looses the context  .. ;)

```
In [11]: data.astype(DecimalDtype(ctx)).dtype
Out[11]: DecimalDtype(context=Context(prec=5, rounding=ROUND_HALF_EVEN, Emin=-999999, Emax=999999, capitals=1, clamp=0, flags=[], traps=[InvalidOperation, DivisionByZero, Overflow]))

In [12]: data.astype(DecimalDtype(ctx)).copy().dtype
Out[12]: DecimalDtype(context=Context(prec=28, rounding=ROUND_HALF_EVEN, Emin=-999999, Emax=999999, capitals=1, clamp=0, flags=[], traps=[InvalidOperation, DivisionByZero, Overflow]))
```

Passing throughself.dtype` here should fix it I think:

https://github.com/pandas-dev/pandas/blob/6c0f95225426a3e06f30a5b279bf9c7a650dce57/pandas/tests/extension/decimal/array.py#L149-L150

Reply:
> Passing throughself.dtype` here should fix it I think:

tried this and no dice

Reply:
It's apparently the `context` that needs to be specified, and not the ``dtype``(something that should maybe be fixed though (separately)):

```diff
     def copy(self):
-        return type(self)(self._data.copy())
+        return type(self)(self._data.copy(), context=self.dtype.context)
```